src/components/App/App.js

```diff
@@ -9,85 +8,313 @@ import Footer from '../Footer/Footer';
 import background from '../../images/georgia-de-lotz--UsJoNxLaNo-unsplash.png';
 import Register from '../Register/Register';
 import Login from '../Login/Login';
+import newsApi from '../../utils/NewsApi';
+import * as auth from '../../utils/auth';
+import { CurrentUserProvider } from '../../contexts/CurrentUserContext';
+import mainApi from '../../utils/MainApi';
+import PopupSuccess from '../PopupSuccess/PopupSuccess';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
+import './App.css';
 
 const App = () => {
+  const history = useHistory();
+
   const [isLoggedIn, setIsLoggedIn] = useState(false);
-  const [preloader, setPreloader] = useState(false);
-  const [isSignupOpen, setIsSignupOpen] = useState(true);
+  const [isRegistered, setIsRegistered] = useState(false);
+  const [isSignupOpen, setIsSignupOpen] = useState(false);
   const [isSigninOpen, setIsSigninOpen] = useState(false);
+  const [preloader, setPreloader] = useState(false);
+  const [placeholder, setPlaceholder] = useState('Enter Topic');
+  const [showNothingFound, setShowNothingFound] = useState(false);
+
+  const [isSuccessPopupOpen, setIsSuccessPopupOpen] = useState(false);
+
+  const [showSearchResults, setShowSearchResults] = useState(false);
+
+  const [showServerError, setShowServerError] = useState(false);
+
+  const [currentKeyword, setCurrentKeyword] = useState(
+    localStorage.getItem('keyword')
+  );
+
+  const [currentUser, setCurrentUser] = useState(null);
+  const [savedArticles, setSavedArticles] = useState([]);
+
+  const [userToken, setUserToken] = useState(localStorage.getItem('token'));
+  // localStorage.removeItem('articles');
+
+  const [articles, setArticles] = useState(
+    JSON.parse(localStorage.getItem('articles'))
+  );
+
   const isMain = usePathname() === '/';
 
-  const handleSignupClick = () => {
-    setIsSignupOpen(true);
+  const checkAuth = () => {
+    auth
+      .checkToken(userToken)
+      .then((res) => {
+        setCurrentUser(res.name);
+        setIsLoggedIn(true);
+      })
+      .catch((err) => {
+        console.log(err);
+        setIsLoggedIn(false);
+      });
+
+    if (isLoggedIn) {
+      mainApi.getUserArticleCollection(userToken).then((res) => {
+        setSavedArticles(res.reverse());
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**